src/main/java/frc/robot/subsystems/TiltedElevatorSubsystem.java

```diff
@@ -0,0 +1,247 @@
+        private double CONE_OFFSET = Units.inchesToMeters(0); // meters, intake jaw to carriage bottom
+
+        /**
+         * ElevatorState defined by absolute height of elevator from ground. All values
+         * in meters and radians.
+         * 
+         * @param absoluteHeight ground to bottom of intake/outtake
+         */
+        private ElevatorState(double extendDistance) {
+            this.extendDistance = extendDistance;
+            this.targetXDistance = 0; // TODO
+        }
+
+        /**
+         * Converts absolute height to extension distance.
+         * 
+         * @param absH absolute height, in meters
+         * @return extension distance, in meters
+         */
+        private static double absHeightToExtendDist(double absHeight) {
+            final double THETA = Math.toRadians(45.0);
+            final double GND_TO_MECH_BOTTOM = Units.inchesToMeters(10.042344); // inches
+
+            return (absHeight - GND_TO_MECH_BOTTOM) / Math.sin(THETA);
+        }
+
+    }
+
+    public TiltedElevatorSubsystem() {
+        extensionMotor = MotorUtil.createSparkMax(EXTENSION_ID);
+
+        extensionMotor.setIdleMode(IdleMode.kCoast); // TODO BRAKE
+
+        extensionMotor.setInverted(true); // flip
+
+        extensionFollowMotor = MotorUtil.createSparkMax(EXTENSION_FOLLOW_ID);
+        extensionFollowMotor.follow(extensionMotor);
+        extensionFollowMotor.setIdleMode(IdleMode.kBrake);
+
+        extensionEncoder = extensionMotor.getEncoder();
+        extensionEncoder.setPositionConversionFactor(EXTENSION_ROT_TO_M);
+        extensionEncoder.setVelocityConversionFactor(EXTENSION_ROT_TO_M / 60.0);
+        extensionEncoder.setPosition(0);
+
+        extensionMotor.enableSoftLimit(SoftLimitDirection.kForward, true);
+        extensionMotor.setSoftLimit(SoftLimitDirection.kForward, EXTENSION_LIMIT);
+        extensionMotor.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        extensionMotor.setSoftLimit(SoftLimitDirection.kReverse, (float) Units.inchesToMeters(-2));
+
+        extensionPidController = extensionMotor.getPIDController();
+        extensionPidController.setP(extensionP);
+        extensionPidController.setI(extensionI);
+        extensionPidController.setD(extensionD);
+        extensionPidController.setFF(extensionFF);
+        extensionPidController.setSmartMotionMaxVelocity(maxVel, 0);
+        extensionPidController.setSmartMotionMaxAccel(maxAccel, 0);
+        extensionPidController.setSmartMotionAccelStrategy(AccelStrategy.kTrapezoidal, 0);
+
+    }
+
+    @Override
+    public void periodic() {
+
+        if (zeroLimitSwitch.get()) {
+            this.extensionEncoder.setPosition(0);
+        }
+        //if at ground state and limit switch has not been hit, get there
+        if (!zeroLimitSwitch.get() && currentState == ElevatorState.GROUND && extensionEncoder.getPosition() <= 0 && offsetDist == 0){
+            extensionMotor.set(-.2);
```

Comment:
this seems to be redundant with lines 186-194, they're different solutions to the same problem